packages/statsd/src/test/client.test.ts

```diff
@@ -0,0 +1,262 @@
+        1,
+        tags,
+        expect.any(Function),
+      );
+    });
+
+    it('passes increment metrics with snake case name metrics to the statsd client when snakeCase=true', () => {
+      const statsDClient = new StatsDClient({
+        ...defaultOptions,
+        snakeCase: true,
+      });
+      statsDClient.increment(stat, tags);
+
+      const stats = StatsDMock.mock.instances[0];
+      const incrementFn = stats.increment;
+      expect(incrementFn).toHaveBeenCalledTimes(1);
+      expect(incrementFn).toHaveBeenCalledWith(
+        stat,
+        1,
+        // eslint-disable-next-line babel/camelcase
+        {foo_bar: tagValue},
+        expect.any(Function),
+      );
+    });
+
+    it('calls errorHandler on the promise when the client returns an error', () => {
+      const errorHandler = jest.fn();
+      const statsDClient = new StatsDClient({...defaultOptions, errorHandler});
+      const statsDMock = StatsDMock.mock.instances[0];
+
+      const error = new Error('Something went wrong!');
+      statsDMock.increment.mockImplementation(
+        (_name, _value, _tags, callback) => {
+          callback(error);
+        },
+      );
+
+      statsDClient.increment(stat, tags);
+      expect(errorHandler).toHaveBeenCalledWith(error);
+    });
+
+    it("calls logger's log on the promise when the client returns an error and there is no errorHandler in options", () => {
+      function logger() {}
+      logger.log = () => {};
+      const logSpy = jest.spyOn(logger, 'log');
+
+      const statsDClient = new StatsDClient({...defaultOptions, logger});
+      const statsDMock = StatsDMock.mock.instances[0];
+
+      const error = new Error('Something went wrong!');
+      statsDMock.increment.mockImplementation(
+        (_name, _value, _tags, callback) => {
+          callback(error);
+        },
+      );
+
+      statsDClient.increment(name, tags);
+      expect(logSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('close', () => {
+    it("calls statsd client's close function", () => {
+      const statsDClient = new StatsDClient(defaultOptions);
+      statsDClient.close();
+
+      const stats = StatsDMock.mock.instances[0];
+      const closeFn = stats.close;
+      expect(closeFn).toHaveBeenCalledTimes(1);
+      expect(closeFn).toHaveBeenCalledWith(expect.any(Function));
+    });
+
+    it('calls errorHandler on the promise when the client returns an error', () => {
+      const errorHandler = jest.fn();
+      const statsDClient = new StatsDClient({...defaultOptions, errorHandler});
+      const statsDMock = StatsDMock.mock.instances[0];
+
+      const error = new Error('Something went wrong!');
+      statsDMock.close.mockImplementation(callback => {
+        callback(error);
+      });
+
+      statsDClient.close();
+      expect(errorHandler).toHaveBeenCalledWith(error);
+    });
+
+    it("calls logger's log on the promise when the client returns an error and there is no errorHandler in options", () => {
```

Comment:
Personally I'd just🔥 the tests that test what we log since we generally don't test logs in other libraries.